app/controllers/nonsipp/CheckReturnDatesController.scala

```diff
@@ -77,11 +79,40 @@ class CheckReturnDatesController @Inject()(
           .bindFromRequest()
           .fold(
             formWithErrors => Future.successful(BadRequest(view(formWithErrors, viewModel))),
-            value =>
-              for {
-                updatedAnswers <- Future.fromTry(request.userAnswers.set(CheckReturnDatesPage(srn), value))
-                _ <- saveService.save(updatedAnswers)
-              } yield Redirect(navigator.nextPage(CheckReturnDatesPage(srn), mode, updatedAnswers))
+            value => {
+
+              val updatedUserAnswers = request.userAnswers.get(CheckReturnDatesPage(srn)) match {
```

Comment:
Quite a bit of logic added here... 
- would it be useful it this is moved to a separate method?
- do we have tests for this logic?

Reply:
I've shrunk the logic quite a bit here now and added unit tests